### ISSUE TITLE
sub-90-degree agent rotation

### ISSUE THREAD

**Issue by zyzhang1130** [allenai/ai2thor#370]
Hi, I would like to check if it is possible for agent to do rotation ( RotateRight/Left) by lest than 90 degrees (say 45 degrees) at moment?

Thank you.

**Comment by winthos** [comment]
The current release does not allow rotating in increments except for 90 degrees via the Rotate actions. Currently you can use the `TeleportFull` action and pass in a rotation that is less than 90 to get a different degree increment. A future release will allow completely free rotation via the Rotate actions.

**Comment by ekolve** [comment]
zyzhang1130,

In the call to the Controller() constructor, try passing in rotateStepDegrees=<value less than 90>

```python
c = Controller(rotateStepDegrees=45.0)
```

**Comment by zyzhang1130** [comment]
@ekolve I tried 'self.event = self.controller.step(rotateStepDegrees=45.0)' got '*** ValueError: Invalid action:'  and 'self.event = self.controller.step(rotateStepDegrees=45.0) got '*** TypeError: 'Controller' object is not callable'. Please advice. Thanks.

**Comment by winthos** [comment]
The `rotateStepDegrees` param doesn't actually function in this way in the current release. Please use the `TeleportFull` action to rotate instead.

**Comment by zyzhang1130** [comment]
Noted. Thanks.

**Comment by ekolve** [comment]
Here is a working example, this is available with the latest release (2.3.5).  Note that ```rotateStepDegrees``` is a parameter to the constructor and not passed to the step() method.  The caveat to this is that the value you pass in to rotateStepDegrees must divide evenly into 360.  If you need a different increment please use TeleportFull as @winthos recommended:

```python
import ai2thor.controller

c = ai2thor.controller.Controller(scene='FloorPlan28', rotateStepDegrees=30.0)
event = c.step(action='RotateRight')
print(event.metadata['agent']['rotation']['y'])
event = c.step(action='RotateRight')
print(event.metadata['agent']['rotation']['y'])
event = c.step(action='RotateRight')
print(event.metadata['agent']['rotation']['y'])
```

**Comment by winthos** [comment]
As an update to this issue, we now support rotating sub 90 degree increments with the latest release, version 2.4.0
